drivers/td-req.c

```diff
@@ -770,6 +771,8 @@ tapdisk_xenblkif_make_vbd_request(struct td_xenblkif * const blkif,
         tapreq->prot = PROT_READ;
         vreq->op = TD_OP_WRITE;
         break;
+    case BLKIF_OP_FLUSH_DISKCACHE:
```

Comment:
Should we also forward this to the underlying storage? The documentation at [https://docs.xenserver.com/en-us/xenserver/8/developer/ddk implies](https://github.com/xen-project/xen/blob/master/xen/include/public/io/blkif.h#L551-L554) implies that this is stronger than a barrier at the tapdisk level: we should also seek some confirmation from the actual underlying storage (O_DIRECT is not enough for that, it can in turn be still cached in volatile storage controller cache).

Reply:
There's not really any mechanism to do that, there is no IO associated with a BLKIF_OP_FLUSH_DISKCACHE so nothing that e.g. an FUA could be attached to.

Reply:
Libaio doesn't support sending an FUA or equivalent. I guess you could do something in the completion path to call `SG_IO ioctl` or `fsync` depending on whether it's block or file based but that's probably a bit complicated.